### PR TITLE
dbg-attr-reg-rw branch

### PR DESCRIPTION
**Pull request by SrikanthPagadarai**

### CODE REVIEW

File: +adi/+common/RegisterReadWrite.m [hunk 1]
```diff
@@ -0,0 +1,38 @@
+classdef (Abstract) RegisterReadWrite < matlabshared.libiio.base 
+    
+    methods (Hidden, Access = protected)
+        function setRegister(obj,settings)
```
Comment (tfcollins) [inline +adi/+common/RegisterReadWrite.m]: What is the input to this function look like?

File: +adi/+common/DebugAttribute.m [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    
+    methods (Hidden, Access = protected)
+        function setDebugAttributeLongLong(obj,attr,value)
+            phydev = getDev(obj, obj.phyDevName);
+            status = iio_device_debug_attr_write_longlong(obj,phydev,attr,value);
+            cstatus(obj,status,['Attribute write failed for : ' attr ' with value ' num2str(value)]);
+            % Check
+            [status, rValue] = iio_device_debug_attr_read_longlong(obj,phydev,attr);
+            cstatus(obj,status,['Error reading attribute: ' attr]);
+            if nargin<6
```
Comment (tfcollins) [inline +adi/+common/DebugAttribute.m]: This condition doesn't make sense

File: +adi/+common/DebugAttribute.m [hunk 3]
```diff
@@ -0,0 +1,36 @@
+classdef (Abstract) DebugAttribute < matlabshared.libiio.base 
+    
+    methods (Hidden, Access = protected)
+        function setDebugAttributeLongLong(obj,attr,value)
```
Comment (tfcollins) [inline +adi/+common/DebugAttribute.m]: Should a tolerance be provided too?

**Comment by tfcollins** [comment]
- Can you add a testcase so into AD9361Tests so others know how to use the AGCSettings
- Can you also sign the commit

**Comment by SrikanthPagadarai** [comment]
>      Can you also sign the commit
I made a couple of updates. I also signed the commits.

**Comment by tfcollins** [comment]
Thanks for making the updates. Just waiting on CI.

**Pull request merged**